### PR TITLE
ci: docker publish edge version

### PR DESCRIPTION
**Pull request by laurentsenta**
Contributes to https://github.com/testground/testground/issues/1430

- Add release workflows to publish docker images for `testground:edge` and `sidecar:edge`
- Add release workflow to build `testground:edge` build artifact as a workflow output
- Add `install.sh` script you may use to automatically setup Testground on your machine
- Update `kind` version and tweaks the env to fix k8s tests in CI

## Tasks

- [x] merge https://github.com/testground/sync-service/pull/13
- [x] add publish docker images on merge to `master`
- [x] add a workflow that outputs the latest master as a build artifact
- [x] add `install.sh` 
- [x] update makefile & code with new container tags
- [x] disable config changes for this test & make envs permanent (use iptestground dockerhub)

### CODE REVIEW

**Comment by laurentsenta** [comment]
Corresponding run: https://github.com/laurentsenta/testground/actions/runs/2934102674

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+docker pull iptestground/sidecar:edge
+
+# At the moment this is the fastest way to get a pre-built testground binary.
+docker run -v ${PWD}:/mount --rm --entrypoint cp iptestground/testground:edge /testground /mount/testground
+
+if [ -z $GITHUB_PATH ]; then
+    echo "Testground was installed to ~/testground. Add this to your path."
+else
+    echo "~/testground" >> $GITHUB_PATH
+fi
```
Comment (laurentsenta) [inline install.sh:18]: @galargh I would like to have this script in the `testground/testground` repo so we can maintain it and improve it without touching the clients. wdyt?

File: .github/workflows/testground-publish-edge.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    steps:
+    - name: checkout
+      uses: actions/checkout@v3
+    - name: Setup Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: "1.16.x"
+        cache: true
+    - name: Build Testground
+      run: go build -ldflags "-X github.com/testground/testground/pkg/version.GitCommit=`git rev-list -1 HEAD`" .
```
Comment (galargh) [inline .github/workflows/testground-publish-edge.yml:24]: Wouldn't that be the same as:
```suggestion
      run: go build -ldflags "-X github.com/testground/testground/pkg/version.GitCommit=${GITHUB_SHA}" .
```

I think it should work for cron triggered workflow run too.

**Review by galargh — APPROVED** [review]
Sweet 🍨

File: install.sh (hunk 1 above)
Comment (galargh) [reply install.sh:18]: Sounds good :)

**Review by brdji — APPROVED** [review]
:+1:

**Comment by laurentsenta** [comment]
Thanks for the review @brdji @galargh!
FYI, I updated the kind version which caused every CI run to break recently.

**Pull request merged**